### DISCUSSION TITLE
installer not coming up

### DISCUSSION THREAD

**Discussion by 14paavang** [JonathanTreffler/backblaze-personal-wine-container#161] [Q&A]
![image](https://github.com/JonathanTreffler/backblaze-personal-wine-container/assets/81815334/e9391100-1e45-473a-968d-642ed45a41e1)
When I get onto the web interface, i'm met with this screen and the installer doesn't boot up

also it did boot up once, but the password filed was missing after the email part, which I currently can not replicate.

**Comment by 14paavang** [comment]
i got v1.5 to work, however, the drive isnt showing up, the software sees drive c, and z for some reason but not d or anything else i map to it

**Reply by 14paavang** [reply]
@traktuner ?

**Reply by traktuner** [reply]
Hey!
Normally, when you use the latest version, the drives should be auto-mapped.
Can you please post the docker logs of the container and the start app-log ( which you can find under `config/dosdevices/c:/backblaze-wine-startapp.log`)

**Reply by 14paavang** [reply]
[backblaze-wine-startapp.log](https://github.com/user-attachments/files/15524578/backblaze-wine-startapp.log)
![image](https://github.com/JonathanTreffler/backblaze-personal-wine-container/assets/81815334/4a01a0a7-2485-476d-baa9-0d1ecee4276e)
it is also detecting and backing up a drive z
after it was not working, i looked at the troubleshooting steps for black screen, and went to the listed errors and saw that v.15 worked, so i installed v.15 and then got it to work, but backblaze was not picking up the drive i had mapped, so i redid the same container with the same volumes mapped, but with the latest image instead, and backblaze then picked the mapped drive up. Then i deleted the old backup that i had made on backblaze( in the backblaze dashboard), and then clicked back up now in the web interface and waited, however a new backup doesn't seem to be showing up. 
( i have mapped my volume as drive g)

**Reply by 14paavang** [reply]
also here are all my drive mappings and environment variables: 
<img width="428" alt="Screenshot 2024-06-02 135558" src="https://github.com/JonathanTreffler/backblaze-personal-wine-container/assets/81815334/a1c8c42d-5824-420b-8c67-c2681b80b31e">
<img width="360" alt="Screenshot 2024-06-02 135512" src="https://github.com/JonathanTreffler/backblaze-personal-wine-container/assets/81815334/ef22c38c-cb45-4b96-857e-ff46cb7ff927">
<img width="437" alt="Screenshot 2024-06-02 135430" src="https://github.com/JonathanTreffler/backblaze-personal-wine-container/assets/81815334/e67e46b2-885c-48d5-8c50-e2394cefd7ba">
<img width="368" alt="Screenshot 2024-06-02 135400" src="https://github.com/JonathanTreffler/backblaze-personal-wine-container/assets/81815334/cad09787-3cdf-4803-b125-8f9b13339924">

**Reply by 14paavang** [reply]
edit: the software told me that i should delete it as the backup had been deleted so I reinstalled the container after wiping the config directory on the host and now even on v.15, the same issue is occuring where the backblaze app isn't coming up in the web interface 
[backblaze-wine-startapp.log](https://github.com/user-attachments/files/15524865/backblaze-wine-startapp.log)

**Reply by 14paavang** [reply]
now i tried reinstalling the image and it has been stuck on the wine config for a bit, also the dosdevices wine folder is empty:
<img width="730" alt="image" src="https://github.com/JonathanTreffler/backblaze-personal-wine-container/assets/81815334/c0425923-dbd1-47cd-8ee8-fa03381d3b1a">
![image](https://github.com/JonathanTreffler/backblaze-personal-wine-container/assets/81815334/9a3025c8-e2d6-4bc2-9173-2b9878fbc1b1)

**Reply by 14paavang** [reply]
apparently what seems to work for me is to install a container with v.15 to set up the config properly and then change the image out to the latest one where the volume mappings haven't been changed so it gets the config from v1.5

**Reply by traktuner** [reply]
hey,
please try to use the tag ubuntu22 instead of latest.
Also the mount screenshot is cut off - for the drive_g mount "read only" is not checked, right?

**Reply by 14paavang** [reply]
> hey, please try to use the tag ubuntu22 instead of latest. Also the mount screenshot is cut off - for the drive_g mount "read only" is not checked, right?

not checked, i'll try ubuntu22

**Reply by 14paavang** [reply]
@traktuner 
I tried a fresh container with ubuntu 22 and it is only a black screen when logging into the web interface, despite giving it root access and every permission.

The other container I have running where I initially installed 1.5 and then upgraded to latest is still working, however, i was backing up my laptop the server in a way that created an image file, but it stopped backing it up after only 2gigabytes of the 120 gigabyte file was backed up. Could this be a result of sending the file over to the server using samba and backing it up at the same time?

**Reply by traktuner** [reply]
Hey,
the folder "dosdevices" should be empty, at least it has content only visible from within the linux container.
The installer files should be in your `config/wine/drive_c` directory - could you please check the permissions on that folder?
If you have 2 containers, one running one not working, they also must not point to the same "config" directory.
Please check again and always post the docker logs and startup.log file so we can debug further. Without the logs we are "blind".
Thank you!

**Reply by 14paavang** [reply]
> Hey, the folder "dosdevices" should be empty, at least it has content only visible from within the linux container. The installer files should be in your `config/wine/drive_c` directory - could you please check the permissions on that folder? If you have 2 containers, one running one not working, they also must not point to the same "config" directory. Please check again and always post the docker logs and startup.log file so we can debug further. Without the logs we are "blind". Thank you!

I purposely mounted a different directory for the ubuntu 22 image config. The only way I could get the latest image to work was by first installing v1.5 and then pointing the latest one to the same config directory. Due to this observation, I reasoned that the latest image wasn't doing something right when setting up the files in its config directory, so I decided to give the ubuntu 22 container a seperate folder mounted as config to see if I was right, and it does seem to be the case. Keep in mind that I explicitly passed the root Id to both containers during creation and I made sure that the ACL on both directories was the same and gave the container full control over everything, explicitly stating so by adding every user and group to it and setting their permissions to full control.

The startapp.log from the ubuntu 22 image is : 
[backblaze-wine-startapp.log](https://github.com/user-attachments/files/15568337/backblaze-wine-startapp.log)
 these logs are from the path you said
[backblaze-wine-startapp.log](https://github.com/user-attachments/files/15568322/backblaze-wine-startapp.log)
These logs are from within the dosdevices folder ( idk why there are 2 copies of it, but I put them both here just in case)

Here are the container logs from the ubuntu22 image: 
[backblaze_backblaze-ix-chart-75bb79b876-s6rnj_ix-chart.log](https://github.com/user-attachments/files/15568365/backblaze_backblaze-ix-chart-75bb79b876-s6rnj_ix-chart.log)


Also the issue I was facing earlier where backblaze wasn't backing up the entire file went away, there seems to be a detection interval or something where the software refuses to back up files it has already backed up, unless they have changed by a significant amount, especially when the file is very large.


Also the dosdevices folder is definitely not empty, it has never been so on any of the containers I have installed ( atleast when viewed from the side of the server (i.e after sharing the entire config directory over samba as Truenas doesn't directly let you view the files stored on it))
![image](https://github.com/JonathanTreffler/backblaze-personal-wine-container/assets/81815334/37af4cd1-6480-4c97-b78b-fecd1ebfbfd8)
this is what is inside the folder. ( this is where I saw the second copy of the startapp logs)

**Reply by Squazz** [reply]
@14paavang did you ever get it working?

**Reply by 14paavang** [reply]
> @14paavang did you ever get it working?

Yes, it worked when I installed A version 1.5 container and then switched the tag to the latest one, making it so that it updated from 1.5 so it retained the old directory setup from that one hut the new app, which seems to be the most stable and best working solution by far, though it’s been so long I  don’t remember of I updated The original container or installed a new one and put it on the same directories, anyway with truenas’ recent switch to docker from kubernetes, the whole process is quite different now, as far as I can tell, vanilla installs should really just be plug and play
